src/jit/lsra.cpp

```diff
@@ -5667,16 +5670,30 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
+        RegRecord* physRegToUnassign = availablePhysRegInterval;
+#ifdef _TARGET_ARM_
+        if (isAssigned(availablePhysRegInterval, currentInterval->registerType))
+#else
         if (intervalToUnassign != nullptr)
+#endif
```

Comment:
There are still a lot of `#ifdef`s in this large method, and I think the above can be simplified. We don't really need to check whether there is an interval to unassign.
At this point, if `intervalToUnassign` is non-null, it is the same as `availablePhysRegInterval->assignedInterval`. So I think the right thing to do here is to unconditionally call a variant of `unassignPhysReg()` that always unassigns the currently assigned interval, if any. Similar to my suggestion above, this method could take an `ARM_ONLY_ARG(RegisterType regType)` and then all this code could be free of `#ifdef`s.